tools/swift-inspect/Sources/SwiftInspectLinux/ElfFile.swift

```diff
@@ -0,0 +1,184 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import LinuxSystemHeaders
+
+class ElfFile {
```

Comment:
You might consider making this generic on something like `<T: ElfEhdr>` and having a small wrapper function to decide which one to instantiate. That would let you remove the `isElf64 ? ... : ...` checks and the use of existentials for things like `ElfSym`, which might be marginally more efficient and nicer to read. Just an idea, definitely not mandatory here.